python/lsst/meas/extensions/scarlet/scarletDeblendTask.py

```diff
@@ -619,24 +619,22 @@ def deblend(self, mExposure, sources):
                 # that the peaks stay consistent
                 for k in skipped:
                     sources.insert(k, None)
-            # Catch the errors that we know about
-            except IncompleteDataError:
-                blendError = "IncompleteDataError"
-            except ScarletGradientError as e:
-                blendError = "ScarletGradientError"
-                src.set(self.iterKey, e.iterations)
-            # Catch all unknown errors to prevent the taks from crashing,
-            # but store the result
+            # Catch all errors and filter out the ones that we know about
             except Exception as e:
-                if self.config.catchFailures:
+                blendError = type(e).__name__
+                if isinstance(e, ScarletGradientError):
+                    src.set(self.iterKey, e.iterations)
+                elif not isinstance(e, IncompleteDataError):
                     blendError = "UnknownError"
```

Comment:
Why override blendError instead of setting it to e.g. `f'{Unexpected {blendError}'`?

Reply:
2 reasons:
1. I don't know the length of the name of the unexpected error that is raised, and unlike numpy the stack isn't smart enough to just chop off a string that is too long, so if the raised error is longer than the number of characters I've allotted for that column in the table it will raise an Exception. I could use `error = ("UnknownError:" + blendError)[:25]` except for the next issue...

2. I want to be able to search for `catalog["deblend_error"] == "UnknownError"` to filter for sources with unknown errors. It is true that `(catalog["deblend_error"] != "ScarletGradientError") & (catalog["deblend_error"] != "IncompleteDataError")` would also work, but as time goes on I expect that we will identify more known failure modes and I'd prefer to not have to change all of my scripts each time. No matter what I'll want to investigate all of the new "UnknownError" blends, so I'm good with this the way it is.